torch/_dynamo/guards.py

```diff
@@ -1128,6 +1129,16 @@ def convert(size_or_stride):
                     log_only=True,
                 )
 
+                # TODO: Figure out if this is the right place for this??
```

Comment:
cc @voznesenskym Is this the correct place for guards on tensor subclass `ctx`? I don't think it's possible to do these from C++ so I guess there's an expected divergence here.

Reply:
A better location would be `TENSOR_MATCH` - but I do wonder if there isn't a way to do this in C++? Not urgent, but we should think about this at some point, all guars are moving to C++ eventually.

Reply:
alright, adding a TODO then